### PR TITLE
SCC-4384: My Account 2.0 VQA/a11y round 1

### PR DESCRIPTION
**Pull request by 7emansell**
## Ticket:

- JIRA ticket [SCC-4384](https://newyorkpubliclibrary.atlassian.net/browse/SCC-4384)

## This PR does the following:

- Addresses comments in [My Account 2.0](https://docs.google.com/document/d/1Rwi0eSEVBEfAMK1ypQ0NkGNd3FtI0ngRAaD-ONPIEvo/edit?tab=t.0) VQA and accessibility review
    - Mainly spacing (VQA) and focus behavior (a11y)

### Question/request for help: 
On mobile when you click edit for Home library/ Notification preference/Password, the fields jump up slightly like they're losing padding/margin or height is just not the same between the two states. I would love a fresh pair of eyes on this since I'm having trouble tracking down why this is happening

## Accessibility concerns or updates

- Adds focus management to all form fields

### Checklist:

- [x] I updated the CHANGELOG with the appropriate information and JIRA ticket number (if applicable).
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

[SCC-4384]: https://newyorkpubliclibrary.atlassian.net/browse/SCC-4384?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/MyAccount/Settings/PasswordForm.tsx
Comment (7emansell) [inline src/components/MyAccount/Settings/PasswordForm.tsx:259]: Vera said: Does this shift the focus to the text input? If so, there actually should be two different refs, just to make it explicit what is being focused on.

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (7emansell) [inline src/components/MyAccount/Settings/SettingsInputForm.tsx:56]: @charmingduchess: 
> With three text inputs open for the phone settings, this focus is triggering on every keystroke input. ie Typing a single character into input 2 jumps the focus to input 3. I think you want an empty array to only run on the first render. I think this will make typescript unhappy but that's ok.

Comment (7emansell) [reply src/components/MyAccount/Settings/SettingsInputForm.tsx:56]: updated it to `tempInputs.length` which I think is right because then it updates only when a new input is added/removed

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (7emansell) [inline src/components/MyAccount/Settings/SettingsInputForm.tsx:214]: @charmingduchess: 

> Not sure if you came across [this ("How to manage a list of refs using a ref callback")](https://react.dev/learn/manipulating-the-dom-with-refs), but your solution is react.dev approved! Looking at their solution, though, I think it might be important to remember to delete refs from the array once the input is removed from the dom.

I tried adding 5 inputs, then deleting one from the middle, and the count on the array of input refs actually went up to 6. It doesn't seem to cause any problems, but I think to avoid bugs down the line, the inputRefs array should match the inputs that are actually present on the dom.

Actually just encountered a bug! after opting to stay logged in from the timed logout modal, the focus is lost and the last input is null, which leads to a loss of focus. I think this can be avoided by removing refs as the inputs are removed.

Comment (7emansell) [reply src/components/MyAccount/Settings/SettingsInputForm.tsx:214]: Updated to now remove ref (L104). Discussed offline but the issue with modal focus isn't going to be addressed here

**Comment by 7emansell** [comment]
Got a new ticket number so closed the old PR and copied Vera's comments over

File: src/components/MyAccount/Settings/PasswordForm.tsx
Comment (7emansell) [reply src/components/MyAccount/Settings/PasswordForm.tsx:259]: @EdwinGuzman this is what we discussed– basically now I'm using 1 ref for everything (except in `SettingsInputForm`, which uses an array of refs to match the array of inputs, and can't really be combined with a button ref). This makes it confusing to predict where the focus will go just from reading the code, but it does work, so I'm torn.

Comment (EdwinGuzman) [reply src/components/MyAccount/Settings/PasswordForm.tsx:259]: For readability and debugging where focus goes, it seems like having two refs is the approach to follow. When reading `useRef<HTMLButtonElement | TextInputRefType | null>`, I was thinking that ref is doing too many things. I think we discussed one ref where possible and updating the `ref.current` value but that's not the case here.

Comment (EdwinGuzman) [reply src/components/MyAccount/Settings/PasswordForm.tsx:259]: To be clear, what other problems are there?

File: src/components/MyAccount/Settings/UsernameForm.tsx
Comment (EdwinGuzman) [inline src/components/MyAccount/Settings/UsernameForm.tsx:172]: From the comment in `PasswordForm` regarding breaking up the ref into two refs, doing so will allow this `as React.Ref...` to be removed.

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (EdwinGuzman) [reply src/components/MyAccount/Settings/SettingsInputForm.tsx:56]: The linter is saying that `focusLastInput` should also go in here. I'm not sure that's right but if it should go here, then the function should be wrapped in a useCallback.

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (EdwinGuzman) [inline src/components/MyAccount/Settings/SettingsInputForm.tsx:231]: I'm not a fan of these magic numbers (72px on line 252). I don't have any good suggestions right now but I'll take a closer look.

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (EdwinGuzman) [inline src/components/MyAccount/Settings/SettingsInputForm.tsx:52]: I was going to ask if it was possible to use the `Map` function like the code example in the React docs resource. It'll work but I imagine this part here would make it difficult to keep track of the previous element to focus?

File: src/components/MyAccount/Settings/PasswordForm.tsx
Comment (charmingduchess) [reply src/components/MyAccount/Settings/PasswordForm.tsx:259]: I think the issue is just conflicting directives. But I think me and Edwin agree this is a case where we can opt for readability.

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (charmingduchess) [reply src/components/MyAccount/Settings/SettingsInputForm.tsx:52]: With an Array we get the order maintained for free, which is crucial here. FWIW, the docs do say "This lets you maintain your own _array_ or a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)".

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (charmingduchess) [reply src/components/MyAccount/Settings/SettingsInputForm.tsx:56]: hmm yes I think the most idiomatic way of doing this is to make the `focusLastInput` a `useCallback` with a dependency of `tempInputs.length`. I'm not too concerned about it. However, I think it is good to see that the `useEffect` is going to run when that value updates, instead of having to check a `useCallback` for a dependency array.

File: src/components/MyAccount/Settings/PasswordForm.tsx
Comment (7emansell) [reply src/components/MyAccount/Settings/PasswordForm.tsx:261]: All refs now split up for readability again 👍

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (7emansell) [reply src/components/MyAccount/Settings/SettingsInputForm.tsx:231]: Same– there are a lot of random fixed values in the designs which prevent using spacing tokens the way I'd like to

File: src/components/MyAccount/Settings/SaveCancelButtons.tsx
Comment (charmingduchess) [reply src/components/MyAccount/Settings/SaveCancelButtons.tsx]: was there a change lost here? i remember there being some conditional logic in the saveCancelButton component around this. I think we discussed it before, but I think it would be ok to add the extra `inputType="password"` in `PasswordForm.tsx`

**Review by charmingduchess — APPROVED** [review]
Looks good!

File: src/components/MyAccount/Settings/SettingsInputForm.tsx
Comment (EdwinGuzman) [reply src/components/MyAccount/Settings/SettingsInputForm.tsx:231]: Right, I said I'll take another pass but I don't want to hold up pushing this out for another review. Okay to add comments post merge?

**Review by EdwinGuzman — APPROVED** [review]
This addresses all the a11y/vqa issues. The clear X button is an issue that we'll debug on the DS so for now, it won't be included in this update.

**Pull request merged**
